File: 1.cpp
#include <string>
#include <unordered_map>
#include <vector>
using namespace std;

class Solution {
public:
  vector<int> twoSum(vector<int> &nums, int target) {
    unordered_map<int, int> map;
    vector<int> result;
    for (int i = 0; i < nums.size(); i++) {
      int complement = target - nums[i];
      if (map.count(complement) != 0) {
        return {map[complement], i};
      }
      map[nums[i]] = i;
    }
    return {};
  }
};

File: 1.py
class Solution:
    def twoSum(self, nums: list[int], target: int) -> list[int]:
        dict = {}
        i = 0
        for value in nums:
            if target - value in dict:
                return [dict[target - value], i]
            dict[value] = i
            i += 1
